### PR TITLE
Contributor settings, redux

### PR DESCRIPTION
**Pull request by pbiggar**
This isn't everything I wanted from #4397, but I'm now officially bored by this work and I want to move on. I'm planning to come back to this though, as I got some pretty good foundations in place I would like to push them around more.

## Before

Screen Shot 2022-08-25 at 12 16 16 AM

## After

Screen Shot 2022-08-25 at 12 15 52 AM

## Description

The main goal here was to try and make forms reusable and nice. This is still a solid refactoring away from being reusable, but it is much nicer.

I added APIFramework, which is similar to API.res, but it doesn't use a model. Instead, it creates a `Callable` type, which takes a record with the 4 values we need to make an API call.

I started using loading and saving states, and also added icons.The form now loads from the server and initialized when you go there. The form views need refactoring, and the form states too.

I also styled the forms. It looks OK, but much better than it did.

Editor canvas work:
- added /api/private/tunnel GET and POST routes

Refactorings:
- uses the Tea.Html2 module (renamed to Tea.Html) which forces using the Tea.Html.Attrs and Events modules. We now use them everywhere.
- renamed Effect to Intent, which matches what Elmish calls it

I made the modal much bigger - it was 50% it's now 75%, but with an absolute limit of 700px. That keeps it small but not too small.

### CODE REVIEW

File: client/src/components/Settings/SettingsContributing.res [hunk 1]
```diff
@@ -1,68 +1,259 @@
...
-@ppx.deriving(show)
-type rec t = {tunnelHost: Utils.formField}
+// -------------------
+// Tunnel URL form
+// -------------------
 
-let default = {tunnelHost: Utils.defaultFormField}
+// This is a single input box that we save/load via API
+module TunnelHost = {
+  // UX:
```
Comment (StachuDotNet) [inline client/src/components/Settings/SettingsContributing.res:11]:
```suggestion
  // UX TODOs:
```

File: client/src/components/Settings/SettingsContributing.res [hunk 2]
```diff
@@ -1,68 +1,259 @@
...
-    | Tea.Result.Ok(false) => ({tunnelHost: {...s.tunnelHost, error: Some("Invalid url")}}, None)
-    | Tea.Result.Error(e) => (
-        {tunnelHost: {...s.tunnelHost, error: Some(Tea.Http.string_of_error(e))}},
-        None,
+  }
+
+  let init = (clientData: APIFramework.clientData) => API.load(clientData)
+
+  let update = (state: t, msg: msg): (t, Intent.t<'msg>) => {
+    Prelude.Debug.loG("update: " ++ show(state) ++ " " ++ show_msg(msg), "")
```
Comment (StachuDotNet) [inline client/src/components/Settings/SettingsContributing.res:111]: Did you meant to leave this in?

**Review by StachuDotNet — APPROVED** [review]

File: client/src/components/Settings/SettingsContributing.res (hunk 1 above)
Comment (pbiggar) [reply client/src/components/Settings/SettingsContributing.res:11]: These are done. I listed the UX as a description of how it should work

File: client/src/components/Settings/SettingsContributing.res (hunk 2 above)
Comment (pbiggar) [reply client/src/components/Settings/SettingsContributing.res:111]: No thanks.

**Pull request merged**